astronomer/providers/sftp/hooks/sftp.py

```diff
@@ -118,23 +118,20 @@ async def list_directory(self, path: str = "") -> Optional[List[str]]:
         except asyncssh.SFTPNoSuchFile:
             return None
 
-    async def get_file_by_pattern(self, path: str = "", fnmatch_pattern: str = "") -> str:
+    async def get_files_by_pattern(self, path: str = "", fnmatch_pattern: str = "") -> List[str]:
         """
         Returns the name of a file matching the file pattern at the provided path, if one exists
         Otherwise, raises an AirflowException to be handled upstream for deferring
         """
         files_list = await self.list_directory(path)
-
+        matched_files = []
         if files_list is None:
             raise AirflowException(f"No files at path {path} found...")
 
         for file in files_list:
```

Comment:
The variable `matched_files` seems to be a bit far from where it's used to me.

I might try something like 

```python
matched_files = [
    file
    for file in files_list
    if fnmatch(file, fnmatch_pattern)
]
```

or `filter` might work as well

Reply:
`matched_files = [file for file in files_list if fnmatch(file, fnmatch_pattern)]` changed